src/browser/components/LoadingAnimation/LoadingAnimation.jsx

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+import useAnimationEnd from '../../hooks/useAnimationEnd.js';
+
+import LoadingIcon from './LoadingIcon.jsx';
+
+const LOADING_STATE = {
+  INITIALIZING: 'initializing', // animation graphics are hidden
+  LOADING: 'loading', // animation graphics fade in and animate
+  LOADED: 'loaded', // animation graphics fade out
+  COMPLETE: 'complete', // animation graphics are removed from the DOM
+};
+
+const ANIMATION_COMPLETION_DELAY = 500;
+
+function LoadingAnimation({
+  loading = false,
+  darkMode = false,
+  onLoadAnimationComplete = () => {}} // eslint-disable-line no-empty-function
+) {
+  const loadingIconContainerRef = React.useRef(null);
```

Comment:
I think this should be:
```suggestion
  const loadingIconContainerRef = React.createRef();
```

Reply:
`React.createRef()` is primarily for class-based components where you store the reference on the class instance. `React.useRef(...)` is the new hook used with function components. The former recreates the ref each time it is called, the latter returns the previously created ref and so is best for function components.

At least that is my understanding and why I used here as I purposely created these as function components to test the use of hooks. Chatted with Guillermo about doing so before I submitted the PR.

Reply:
yes, I think `createRef` was not supposed to work with hooks, I think I read that somewhere, but I can't seem to find it.

Reply:
Strange that they would duplicate that functionality.